src/posthog-surveys.ts

```diff
@@ -214,34 +221,50 @@ export class PostHogSurveys {
             return
         }
 
-        const generateSurveys = phExtensions.generateSurveys
-
         if (!this._decideServerResponse) {
             logger.warn('Decide not loaded yet. Not loading surveys.')
             return
         }
 
-        if (this._surveyEventReceiver == null) {
-            this._surveyEventReceiver = new SurveyEventReceiver(this.instance)
-        }
-
-        if (!generateSurveys) {
-            const loadExternalDependency = phExtensions.loadExternalDependency
-
-            if (loadExternalDependency) {
-                loadExternalDependency(this.instance, 'surveys', (err) => {
-                    if (err) {
-                        logger.error('Could not load surveys script', err)
-                        return
-                    }
-
-                    this._surveyManager = phExtensions.generateSurveys?.(this.instance)
-                })
+        this._isInitializingSurveys = true
+
+        try {
+            const generateSurveys = phExtensions.generateSurveys
+
+            if (!generateSurveys) {
+                const loadExternalDependency = phExtensions.loadExternalDependency
+
+                if (loadExternalDependency) {
+                    loadExternalDependency(this.instance, 'surveys', (err) => {
+                        if (err) {
+                            logger.error('Could not load surveys script', err)
+                            this._isInitializingSurveys = false
+                            return
+                        }
+
+                        this._surveyManager = phExtensions.generateSurveys?.(this.instance)
+                        this._isInitializingSurveys = false
+                        if (this._surveyEventReceiver == null) {
+                            this._surveyEventReceiver = new SurveyEventReceiver(this.instance)
+                        }
+                        logger.info('Surveys loaded successfully')
```

Comment:
could it be that `phExtensions.generateSurveys` returns null? if so, we have to check if `_surveyManager` has a value and then log its successfully loaded or not

Reply:
this is executed after the callback that should load it into phExtensions.

but I agree we should be more defensive here - so I [added an additional check](https://github.com/PostHog/posthog-js/pull/1747/commits/899bff9c3acc6930a313cbb22005be7bc8794477), before seeing if there's an error, to see if this value is undefined.